src/__tests__/SearchResults.test.js

```diff
@@ -0,0 +1,14 @@
+import React from "react";
+import { getByRole, render, screen } from "@testing-library/react";
+import SearchResults from "../components/SearchResults";
+
+describe('Search Results', () => {
+  const validProps = {
+    results: [],
+  }  
+  const { getByText } = render(<SearchResults results={validProps.results} />);
+
+  test("search results render correctly", () => {
+    expect(screen.getByText(/results/i)).toBeInTheDocument()
+  })
```

Comment:
So a few things here: 
- You should put the render method inside the body the test as that is a thing that almost always needs to happen per test, to keep things separate for all the functionality you are trying to test
- You dont need to to desrtucture the `getByText` method from `render`, only to then get the same method off the screen object as you do on line 12. You could just render the component and remove the destructuring
- This isnt really much a useful test as it is. It passes, but only because you are looking for a value that will always be in the document regardless of there being any results to display or not. If you are passing in an empty array (as you are here), you would expect the message to be "No results" so you should assert for that:

```
  test("search results render correctly", () => {
    render(<SearchResults results={validProps.results} />);
    expect(screen.getByText("No results")).toBeInTheDocument()
  })
```
You now have half of the possible expected outcomes covered by your test. The other is what should be displayed if you pass an array in? Normally we would expect there to be the same number of images as the length of the array of URLs passed in, but your app is a bit more complex with an object being passed in and thumbnails/large images being displayed too, so this would need to be a bit more involved:

```
  test("Correct number of images are rendered with correct attributes", () => {
    validProps = {
      results: [
        {
          description: "test description 1",
          original: "https://test-original-url-1.jpg",
          originalHeight: 600,
          originalTitle: "test title 1",
          thumbnail: "https://test-thumbnail-url-1.jpg"
        },
        {
          description: "test description 2",
          original: "https://test-original-url-2.jpg",
          originalHeight: 800,
          originalTitle: "test title 2",
          thumbnail: "https://test-thumbnail-url-2.jpg"
        },
      ]
    };

    render(<SearchResults results={validProps.results} />)
    const allImages = screen.getAllByRole("img");

    expect(
      screen.getByText(`${validProps.results.length} results`)
    ).toBeInTheDocument();
    expect(allImages).toHaveLength(validProps.results.length * 2);
    expect(allImages[2]).toHaveAttribute("src", "https://test-original-url-1.jpg");
    expect(allImages[3]).toHaveAttribute("src", "https://test-original-url-2.jpg");
  });
```

Once you get the hang of it testing really can be quite powerful and an extremely useful tool to have at your disposal!

Reply:
I think I was that involved in looking at the API and extracting the various bits of information from it and then inserting them into the ImageGallery component and testing it "manually" by checking it rendered correctly visually, that the writing of test code went out the window. That said, i do appreciate the importance of TDD and how it will be essential when doing a "real" tech test when it comes to applying for jobs. Have expanded the PropTypes for this test and test still passes :o)